velox/exec/Aggregate.h

```diff
@@ -139,6 +150,13 @@ class Aggregate {
   // 'groups'. No-op for fixed length accumulators.
   virtual void destroy(folly::Range<char**> /*groups*/) {}
 
+  // Returns the type that extractAccumulator will produce. This is
+  // different from resultType_ for complex final aggregations. Needed
+  // for e.g. spilling.
+  virtual TypePtr accumulatorType() const {
```

Comment:
It is confusing that the comment says "This is different from resultType_", but the implementation returns resultType_. It might be cleaner to not provide the default implementation for this method and require individual aggregations to specify this. Alternatively, Aggregate constructor could take accumulatorType as a parameter.